packages/mvc.php

```diff
@@ -0,0 +1,343 @@
+        }
+
+        $ex->exportTable('Role', "
+            select
+                r.VanillaID as RoleID,
+                m.RoleName as Name
+            from MembershipRole m, RoleId r
+            where r.Id = m.Id
+         ", $role_Map);
+
+        // User Role.
+        $userRole_Map = array();
+        $ex->exportTable('UserRole', "
+            select
+                u.VanillaID as UserID,
+                r.VanillaID as RoleID
+            from MembershipUsersInRoles m, RoleId r, UserId u
+            where r.Id = m.RoleIdentifier and u.Id = m.UserIdentifier
+        ", $userRole_Map);
+
+        //Badge.
+        $badge_Map = array();
+
+        if(!$ex->tableExists("BadgeId")){
+            $ex->query("create table BadgeId as (
+                    select Id from Badge)
+            ");
+
+            $ex->query("ALTER TABLE BadgeId ADD VanillaID INT NOT NULL AUTO_INCREMENT PRIMARY KEY");
+        }
+
+        $ex->exportTable('Badge', "
+            select
+                b.VanillaID as BadgeID,
+                m.Type as Type,
+                m.DisplayName as Name,
+                m.Description as Body,
+                m.Image as Photo,
+                m.AwardsPoints as Points
+            from Badge m, BadgeId b
+            where b.Id = m.Id
+        ", $badge_Map);
+
+        $user_badge_Map = array();
+        $ex->exportTable('UserBadge', "
+            select
+                u.VanillaID as UserID,
+                b.VanillaID as BadgeID,
+                '' as Status,
+                now() as DateInserted
+            from MembershipUser_Badge m, UserId u, BadgeId b
+            where u.Id = m.MembershipUser_Id and b.Id = m.Badge_Id
+        ", $user_badge_Map);
+
+        // Category.
+        $category_Map = array();
+
+        if(!$ex->tableExists("CategoryId")){
+            $ex->query("create table CategoryId as (
+                    select Id from :_Category)
+            ");
+
+            $ex->query("ALTER TABLE CategoryId ADD VanillaID INT NOT NULL AUTO_INCREMENT PRIMARY KEY");
+        }
+
+        $ex->exportTable('Category', "
+
+                          select
+                c.VanillaID as CategoryID,
+                p.VanillaID as ParentCategoryID,
+                m.Name as Name,
+                m.Description as Description,
+                m.DateCreated as DateInserted,
+                null as Sort
+            from Category m, CategoryId c, CategoryId p
+            where m.Category_Id <> '' and c.Id = m.Id and p.Id = m.Category_Id
+
+            UNION
+
+            select
+                c.VanillaID as CategoryID,
+                '-1' as ParentCategoryID,
+                m.Name as Name,
+                m.Description as Description,
+                m.DateCreated as DateInserted,
+                null as Sort
+            from Category m, CategoryId c
+            where m.Category_Id = '' and c.Id = m.Id
+
+
+        ", $category_Map);
```

Comment:
I know old packages aren't always respecting standards and aren't always properly indented, but you should try to avoid leaving blank lines in your queries. Also, I recommend that you stick with lower case SQL keywords.

```suggestion
        $ex->exportTable('Category', "
             select
                c.VanillaID as CategoryID,
                p.VanillaID as ParentCategoryID,
                m.Name as Name,
                m.Description as Description,
                m.DateCreated as DateInserted,
                null as Sort
            from Category m, CategoryId c, CategoryId p
            where m.Category_Id <> '' and c.Id = m.Id and p.Id = m.Category_Id

            union

            select
                c.VanillaID as CategoryID,
                '-1' as ParentCategoryID,
                m.Name as Name,
                m.Description as Description,
                m.DateCreated as DateInserted,
                null as Sort
            from Category m, CategoryId c
            where m.Category_Id = '' and c.Id = m.Id
        ", $category_Map);
```